src/commands/org/open.ts

```diff
@@ -113,14 +133,35 @@ export class OrgOpenCommand extends SfCommand<OrgOpenOutput> {
     return output;
   }
 
-  private async buildFrontdoorUrl(version?: string): Promise<string> {
+  private async buildFrontdoorUrl(): Promise<string> {
     await this.org.refreshAuth(); // we need a live accessToken for the frontdoor url
-    const conn = this.org.getConnection(version);
-    const accessToken = conn.accessToken;
+    const accessToken = this.conn.accessToken;
     const instanceUrl = this.org.getField<string>(Org.Fields.INSTANCE_URL);
     const instanceUrlClean = instanceUrl.replace(/\/$/, '');
     return `${instanceUrlClean}/secur/frontdoor.jsp?sid=${accessToken}`;
   }
+
+  private async generateFileUrl(file: string): Promise<string> {
+    try {
+      const metadataResolver = new MetadataResolver();
+      const components = metadataResolver.getComponentsFromPath(file);
```

Comment:
I think you might want to assert that exactly one component was found and throw otherwise.

components[0] is like an `undefined` bug waiting to happen

Reply:
I love `array.at` but we can't use it until node14 is EOL, >4/30.  You'll probably see UT failing from it.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at